src/libs/export.c

```diff
@@ -213,6 +227,135 @@ void _set_dimensions(dt_lib_export_t *d, int max_width, int max_height)
+{
+    dt_conf_set_int(CONFIG_PREFIX "height", 0);
+    g_signal_handler_block(G_OBJECT(d->height), d->height_chg1_handler_id);
+    g_signal_handler_block(G_OBJECT(d->height), d->height_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(d->height), "0");
+    g_signal_handler_unblock(G_OBJECT(d->height), d->height_chg1_handler_id);
+    g_signal_handler_unblock(G_OBJECT(d->height), d->height_chg2_handler_id);
+
+    dt_conf_set_int(CONFIG_PREFIX "width", 0);
+    g_signal_handler_block(G_OBJECT(d->width), d->width_chg1_handler_id);
+    g_signal_handler_block(G_OBJECT(d->width), d->width_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(d->width), "0");
+    g_signal_handler_unblock(G_OBJECT(d->width), d->width_chg1_handler_id);
+    g_signal_handler_unblock(G_OBJECT(d->width), d->width_chg2_handler_id);
+}
+
+static void size_changed_reset_scale(GtkWidget *spin, dt_lib_export_t *d)
+{
+  char new_value[2] = "1\0";
+  dt_conf_set_string(CONFIG_PREFIX "resizing_factor", new_value);
+  g_signal_handler_block(G_OBJECT(d->scale), d->scale_chg1_handler_id);
+  g_signal_handler_block(G_OBJECT(d->scale), d->scale_chg2_handler_id);
+  gtk_entry_set_text (GTK_ENTRY(d->scale), new_value);
+  g_signal_handler_unblock(G_OBJECT(d->scale), d->scale_chg1_handler_id);
+  g_signal_handler_unblock(G_OBJECT(d->scale), d->scale_chg2_handler_id);
+}
+
+static void scale_mdlclick(GtkEntry *spin, GdkEventButton *event, dt_lib_export_t *d)
+{
+  if (event->button == 2){
+    size_changed_reset_scale(GTK_WIDGET(spin), d);
+  } else {
+    scale_changed(spin);
+  }
+}
+
+static void widht_mdlclick(GtkEntry *spin, GdkEventButton *event, gpointer user_data)
+{
+  if (event->button == 2){
+    dt_lib_export_t *d = (dt_lib_export_t *)user_data;
+    g_signal_handler_block(GTK_ENTRY(spin), d->width_chg1_handler_id);
+    g_signal_handler_block(GTK_ENTRY(spin), d->width_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(spin), "0");
+    g_signal_handler_unblock(GTK_ENTRY(spin), d->width_chg1_handler_id);
+    g_signal_handler_unblock(GTK_ENTRY(spin), d->width_chg2_handler_id);
+    dt_conf_set_int(CONFIG_PREFIX "width", atoi(gtk_entry_get_text(GTK_ENTRY(spin))));
+  } else {
+    width_changed(GTK_EDITABLE(spin), user_data);
+  }
+}
+
+static void height_mdlclick(GtkEntry *spin, GdkEventButton *event, gpointer user_data)
+{
+  if (event->button == 2){
+    dt_lib_export_t *d = (dt_lib_export_t *)user_data;
+    g_signal_handler_block(GTK_ENTRY(spin), d->height_chg1_handler_id);
+    g_signal_handler_block(GTK_ENTRY(spin), d->height_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(spin), "0");
+    g_signal_handler_unblock(GTK_ENTRY(spin), d->height_chg1_handler_id);
+    g_signal_handler_unblock(GTK_ENTRY(spin), d->height_chg2_handler_id);
+    dt_conf_set_int(CONFIG_PREFIX "height", atoi(gtk_entry_get_text(GTK_ENTRY(spin))));
+  } else {
+    height_changed(GTK_EDITABLE(spin), user_data);
+  }
+}
+static void doom_changed(GtkWidget *spin, gpointer user_data)
```

Comment:
doom???

Reply:
I'm sorry, it was a splinter of mental anguish :-(